### PR TITLE
User chatting

### PR DESCRIPTION
**Pull request by abdullah-faisal-qbatch**
Implemented user chat rooms

### CODE REVIEW

File: room/views.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+from django.contrib.auth.models import User
+from django.contrib.auth import login
+from rest_framework import generics
+from django.http import HttpResponse
+from .models import Room
+from django.shortcuts import render, redirect
+from django.contrib.auth.decorators import login_required
+from django.shortcuts import render
+
+from .models import Room, Message
```
Comment (farid-hussain-qbatch) [inline room/views.py]: There are multiple same imports, also it should be in ascending way like this

from dotenv import load_dotenv
from celery import shared_task
from sp_api.api import Reports
from confluent_kafka import Producer
from sp_api.api import Notifications
from datetime import datetime, timedelta

File: room/views.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        print("********************************")
+        name = request.POST['name']
+        slug = request.POST['slug']
+        room = Room(name=name, slug=slug)
+        room.save()
+        return redirect('room/rooms')
+    # Redirect to a room list page
+    # return render(request, 'room/room_form.html')
+
+
```
Comment (farid-hussain-qbatch) [inline room/views.py]: Add docstring for any one or two method to have an idea

File: room/views.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+def rooms(request):
+    all_rooms = Room.objects.all()
+    print(all_rooms)
+    return render(request, 'room/rooms.html', {'rooms': all_rooms})
+
+
+@login_required
+def room(request, slug):
+    room = Room.objects.get(slug=slug)
+    messages = Message.objects.filter(room=room)[:25]
```
Comment (farid-hussain-qbatch) [inline room/views.py]: Why using this [:25] here?

File: instagram/settings.py [hunk 4]
```diff
@@ -32,16 +32,20 @@
 # Application definition
 
 INSTALLED_APPS = [
+    "daphne",
```
Comment (Nabeel-Akhtar-Qbatch) [inline instagram/settings.py:35]: For what purpose , this app is bring used

**Review by farid-hussain-qbatch — APPROVED** [review]

**Pull request merged**